UnityGameBase/Core/Utils/Extensions.cs

```diff
@@ -89,7 +89,12 @@ public static T GetRandomAndRemove<T>(this List<T> target)
         
         public static void Shuffle<T>(this List<T> list)
         {
-            System.Random rng = new System.Random();  
+            // System.Random() initialisation is obviously based on a low
+            // resolution timestamp of the current time, which yields
+            // identical shuffles if called in close succession.
+            // Solution: Use milliseconds, which provide more variation.
+            System.Random rng = new System.Random(System.DateTime.Now.Millisecond);
```

Comment:
Bad or not, it apparently fixes the problem. From observation, the default implementation yields the same pseudorandom numbers so often in called in close succession that I seriously doubt the resolution is that high. Anyway, you have probably seen pull request https://github.com/kwnetzwelt/ugb-source/pull/8 which uses ticks instead of milliseconds.

Making the RNG static sounds like the most reasonable idea. Please go ahead. :-)